spec/models/repository/subversion_spec.rb

```diff
@@ -0,0 +1,274 @@
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+
+describe Repository::Subversion, type: :model do
+  let(:instance) { FactoryGirl.build(:repository_subversion) }
+  let(:adapter)  { instance.scm }
+  let(:config)   { {} }
+
+  before do
+    allow(Setting).to receive(:enabled_scm).and_return(['Subversion'])
+    allow(instance).to receive(:scm).and_return(adapter)
+    allow(adapter).to receive(:config).and_return(config)
+  end
+
+  describe 'default Subversion' do
+    it 'is not manageable' do
+      expect(instance.manageable?).to be false
+    end
+
+    it 'has one available type' do
+      expect(instance.supported_types).to eq [:existing]
+    end
+  end
+
+  describe 'managed Subversion' do
+    let(:managed_path) { '/tmp/managed_svn' }
+    it 'is not manageable unless configured explicitly' do
+      expect(instance.manageable?).to be false
+    end
+
+    context 'with managed config' do
+      let(:config) { { manages: managed_path } }
+      let(:project) { FactoryGirl.build :project }
+      let(:identifier) { "#{project.identifier}.svn" }
+
+      it 'is manageable' do
+        expect(instance.manageable?).to be true
+      end
+
+      it 'has two available types' do
+        expect(instance.supported_types).to eq [:existing, :managed]
+      end
+
+      context 'and associated project' do
+        before do
+          instance.project = project
+        end
+
+        it 'outputs valid managed paths' do
+          expect(instance.repository_identifier).to eq(identifier)
+          path = File.join(managed_path, identifier)
+          expect(instance.managed_repository_path).to eq(path)
+          expect(instance.managed_repository_url).to eq("file://#{path}")
+        end
+      end
+
+      context 'and associated project with parent' do
+        let(:parent) { FactoryGirl.build :project }
+        let(:project) { FactoryGirl.build :project, parent: parent }
+
+        before do
+          instance.project = project
+        end
+
+        it 'outputs the correct hierarchy path' do
+          expect(instance.managed_repository_path)
+            .to eq(File.join(managed_path, parent.identifier, identifier))
+        end
+      end
+    end
+  end
+
+  describe 'with an actual repository' do
+    with_filesystem_repository('subversion', 'svn') do |repo_dir|
```

Comment:
Unused block argument - `repo_dir`. You can omit the argument if you don't care about it.